scripts/controller.gd

```diff
@@ -78,18 +81,24 @@ func _start_round(payload: Dictionary):
 	self._round_number = payload["round_number"]
 	self._timer.start(payload["round_duration"])
 	var pieces = payload["board_state"]
+	self._place_pieces(pieces)
+	self._board.turn_all_player_piece_lights_on(self._selectable_light_intensity)
+
+func _place_pieces(pieces: Array):
 	for piece in pieces:
 		var piece_pos = Position.get_obj_from_dict(piece["position"])
 		self._board.place_piece(piece["piece_id"], piece["player_id"], piece_pos.get_vec())
-	self._board.turn_all_player_piece_lights_on(self._selectable_light_intensity)
 
 func _animate_round(payload: Dictionary):
 	await self._board.animate_events(payload["timeline"])
+	if not payload["game_over"] == null:
+		GlobalVariables.winner_id = payload["game_over"]["winner_player_id"]
+		get_tree().change_scene_to_file("res://scenes/lobby.tscn")
+	self._current_state = STATES.AWAITING_ROUND
```

Comment:
The client still needs to send the ready_for_next_round message here. See https://github.com/dsg-jam/ld51-server/blob/main/docs/protocol.md#game-over

Reply:
18a9ee2